Multisig/UI/App/MainTabBarViewController.swift

```diff
@@ -332,12 +334,29 @@ class MainTabBarViewController: UITabBarController {
     
     private func showTransactionDetails(safeTxHash: Data) {
         let vc = ViewControllerFactory.transactionDetailsViewController(safeTxHash: safeTxHash)
-        present(vc, animated: true, completion: nil)
+        presentWhenReady(vc)
     }
     
     private func showTransactionDetails(transactionDetails: SCGModels.TransactionDetails) {
         let vc = ViewControllerFactory.transactionDetailsViewController(transaction: transactionDetails)
-        present(vc, animated: true, completion: nil)
+        presentWhenReady(vc)
+    }
+
+    private var delayedPresentationViewController: UIViewController?
+
+    private func presentWhenReady(_ vc: UIViewController) {
+        if view.window == nil {
```

Comment:
if app is closed view.window == nil but MainTabBarViewController still receives notificaiton callbacks?

Reply:
this condition means that the controller is not in the window hierarchy, i.e. it is not on screen yet. This happens when the animation is not started. When notification is received, the app is not fully functional, that's why it needs delay until the tab bar is shown